Cython/Compiler/ParseTreeTransforms.py

```diff
@@ -3021,6 +3021,28 @@ def visit_Node(self, node):
             node.in_nogil_context = True
         return node
 
+    def visit_ModuleNode(self, node):
+        # Not strictly a GIL check however check to see if numpy import_array is called
+        # should probably be done as late as possible
```

Comment:
I think this can well be moved to the code generation phase of `CImportStatNode` and `FromCImportStatNode`.